push_notifications/admin.py

```diff
@@ -47,21 +47,31 @@ def send_messages(self, request, queryset, bulk=False):
 		# Because NotRegistered and InvalidRegistration do not throw GCMError
 		# catch them here to display error msg.
 		if not bulk:
-			for r in list(ret):
-				if r in ("Error=NotRegistered", "Error=InvalidRegistration"):
-					errors.append(r)
-					ret.remove(r)
+			for r in ret:
+				if "error" in r["results"][0]:
+					errors.append(r["results"][0]["error"])
 		else:
-			errors = [",".join(r.values()) for r in ret[0][0]["results"] if "error" in r]
+			try:
+				errors = [r["error"] for r in ret[0][0]["results"] if "error" in r]
+			except TypeError:
+				for entry in ret[0][0]:
+					errors = errors + [r["error"] for r in entry["results"] if "error" in r]
 		if errors:
 			self.message_user(
 				request, _("Some messages could not be processed: %r" % (", ".join(errors))),
 				level=messages.ERROR
 			)
 		if ret:
-			if not bulk:
-				ret = ", ".join(ret)
-			elif ret[0][0]["success"] == 0:
+			if bulk:
+				try:
+					success = ret[0][0]["success"]
+				except TypeError:
```

Comment:
When the queryset exceeds the max_recipients value, the send_message method returns a list of dicts, one per chunk.